pkg/controller/machine-set-boot-image/platform_helpers.go

```diff
@@ -105,9 +105,58 @@ func checkMachineSet(infra *osconfigv1.Infrastructure, machineSet *machinev1beta
 	}
 }
 
-func reconcileAWS(machineSet *machinev1beta1.MachineSet, _ *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
-	klog.Infof("Skipping machineset %s, unsupported platform type AWS with %s arch", machineSet.Name, arch)
-	return false, nil, nil
+func reconcileAWS(machineSet *machinev1beta1.MachineSet, configMap *corev1.ConfigMap, arch string) (patchRequired bool, newMachineSet *machinev1beta1.MachineSet, err error) {
+	klog.Infof("Reconciling MAPI machineset %s on AWS, with arch %s", machineSet.Name, arch)
+
+	// First, unmarshal the AWS providerSpec
+	providerSpec := new(machinev1beta1.AWSMachineProviderConfig)
+	if err := unmarshalProviderSpec(machineSet, providerSpec); err != nil {
+		return false, nil, err
+	}
+
+	// Next, unmarshal the configmap into a stream object
+	streamData := new(stream.Stream)
+	if err := unmarshalStreamDataConfigMap(configMap, streamData); err != nil {
+		return false, nil, err
+	}
+	// Extract the region from the Placement field
+	region := providerSpec.Placement.Region
```

Comment:
question: can any part of this ever be nil? There's a possibility of panic if we don't do a nil check, so was just wondering. If this can't be nil (api guarded) then it's fine as is

Reply:
The Region should never be able to be nil so we should be API guarded. If it is nil then the cluster is not in a good state